module/VuFind/src/VuFind/Controller/AbstractSolrSearch.php

```diff
@@ -146,11 +146,9 @@ protected function processAdvancedFacets(
         $hierarchicalFacetsSortOptions = []
     ) {
         // Process the facets
-        $facetHelper = null;
-        if (!empty($hierarchicalFacets)) {
-            $facetHelper = $this->serviceLocator
-                ->get(\VuFind\Search\Solr\HierarchicalFacetHelper::class);
-        }
+        $facetHelper = $this->serviceLocator
+            ->get(\VuFind\Search\Solr\HierarchicalFacetHelper::class);
+        $facetConfig = $this->getConfig('facets');
```

Comment:
I don't think we should hard-code the 'facets' configuration here. This controller can be shared by multiple Solr-based searches, and not all of them use the same configuration files. It would probably be better if we could leverage the appropriate Options object to figure out this behavior -- i.e. add a new getter there. (It should be possible to retrieve a generic Options object based on $this->searchClassId if a specific instance is not passed in via $searchObject).

Reply:
oh yea, makes sense. Hmm have to readjust and if Ere has something to say about this implementation.